### PR TITLE
Fixes #5413 (Newscasters  are no longer floating everywhere)

### LINKED ISSUE #5413: Newscasters on box are no longer on walls
>Revision (Should be above if you're viewing this from ingame!)
c0be34f456787d35f7d09b542f2bb4531966a062 

>General description of the issue
dreamseeker 2015-07-18 14-14-10-988

### PR DESCRIPTION
**Pull request by Intigracy**
Fixes #5413

Newscasters now require screwdriving after placing on the wall, instead of immediately finishing.

Newscasters are now unscrewed and then crowbared to remove from the wall.

Mapping newscasters remains unchanged from before the issue was presented.

Dmi change includes the unscrewdrivered icon / assembly icon, last two in this image.

### CODE REVIEW

**Comment by Exxion** [comment]
I don't think that's what #5413 was about
I believe #5413 is about the ones that are floating at roundstart due to the fixing of shitcode but not shitmapping

**Comment by Intigracy** [comment]
Newscasters are currently mapped the exact same way as every other wall mounted object in the game, via pixel_x and pixel_y editing.

The code that fixed the issue of newscaster construction dropping it right below you (#5301, fixed in #5389) is what led to the floating newscasters everywhere.

This reverts that part of #5389 and fixes #5301 to be in line with other wall mountables.

File: code/game/machinery/newscaster.dm [hunk 1]
```diff
@@ -132,18 +133,16 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
...
-	else
-		dir = reverse_direction(dir) //Unfucks all maps
-	pixel_x = (dir & 3)? 0 : (dir == 4 ? 28 : -28)
-	pixel_y = (dir & 3)? (dir ==1 ? 33 : -28) : 0
+/obj/machinery/newscaster/New(var/loc, var/ndir, var/building = 1)
+	buildstage = building
+	if(!buildstage) //Already placed newscasters via mapping will not be affected by this
+		pixel_x = (ndir & 3)? 0 : (ndir == 4 ? 28 : -28)
+		pixel_y = (ndir & 3)? (ndir == 1 ? 28 : -28) : 0
+		dir = ndir
```
Comment (Intigracy) [inline code/game/machinery/newscaster.dm]: I do want to note that lines 138-141 right here were all that were needed for the original fix, and it wouldn't have messed with pre-mapped ones. I stole this from light switch code.

**Comment by Exxion** [comment]
Wait, they are? So what the fuck even went wrong?
Sorry, last wall mountable code I touched was fireaxe cabinets, which... weren't even mountable at the time

**Comment by Intigracy** [comment]
>+/obj/machinery/newscaster/New(turf/loc, var/ndir=0)
>+	if(ndir)
>+		dir=ndir
>+	else
>+		dir = reverse_direction(dir) //Unfucks all maps
>+	pixel_x = (dir & 3)? 0 : (dir == 4 ? 28 : -28)
>+	pixel_y = (dir & 3)? (dir ==1 ? 33 : -28) : 0

This is what caused the floating ones we see now, see above on my line comment for how it is now.

**Comment by Intigracy** [comment]
At some point I'll bring back ATM / request console creation using this same method.

**Comment by Intigracy** [comment]
@Kurfursten, I'd like to apologize if I came off as an ass, I was just irritated that I opened the map and fixed all the newscasters using what I thought was the right method, only to find out about your dir flipping.

**Comment by Kurfursten** [comment]
It's okay, I'm having kind of a rough time irl right now and probably came off as more defensive than was necessary. This is good work, :+1:

**Pull request merged**